### ISSUE TITLE
Limit of 8 colliders is very limiting.

### ISSUE THREAD

**Issue by methusalah** [Unity-Technologies/com.unity.demoteam.hair#27]
In my game, the hair can be in collision with a lot of different objects, for example :
 - the character skin,
 - another character touching them,
 - player's hands (in VR),
 - any object the player hold in their hand.

These objects can collide one another and only use primitive collider for obvious performance reasons. Therefore, the hair can be in contact with hundreds of spheres simultaneously. Creating SDF for all the objects in game would be a very long and hard work. Updating the SDF of the dynamic ones would also have a significant performance impact.

Also, colliding with hundreds of sphere is often faster than updating and testing a single SDF.

Is there a way to lift the limit of 8 colliders somehow?

**Comment by fuglsang** [comment]
Thank you for the report!

For now the limitation set rather conservatively. When boundaries are not specified specifically, this really acts as an active limit, since boundaries by default are being collected within the hair volume per-frame. I have a small TODO to also prioritize the active boundaries by distance, which unfortunately didn't make it into the initial release.

Pending other more pressing issues, we can take a look at lifting the limitation later. From my perspective, there are some important things to consider in this regard:
- No performance regressions allowed: Increasing the size of the primitive buffer should not negatively impact performance when using only a small amount of primitives.
- It may be interesting to also consider an option of combining a high number of input primitives to discrete SDF + surface velocity volume, at the cost of surface precision during collision resolve.

**Comment by fuglsang** [comment]
> As far as I can tell, the only way you can specify boundaries is in the "Boundaries Priority" list. These boundaries are also affected by the limit of 8 colliders, despite the tooltip stating "Always included boundaries". If I have more than 1 SDF + 7 spheres in this list, some of them will be discarded. I tested it with "Boundaries Collect" on and off with the same result.
> 
> Is it the intended result? Is there another way to specify boundaries and have more than 8 of them at the same time?

Correct. Currently there is no way to have more than 8 active boundaries.

There is a related known issue, however, with boundaries in the priority list not actually taking priority over boundaries automatically gathered via overlap query.
